src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java

```diff
@@ -21,8 +32,10 @@
 import org.apache.qpid.proton.engine.impl.TransportOutput;
 import org.apache.qpid.proton.engine.impl.TransportWrapper;
 
+import javax.xml.bind.DatatypeConverter;
+
 public class ProxyImpl implements Proxy, TransportLayer {
-    private final int proxyHandshakeBufferSize = 4 * 1024; // buffers used only for proxy-handshake
+    private final int proxyHandshakeBufferSize = 2 * 1024 * 1024; // buffers used only for proxy-handshake

```

Comment:
Why is the buffer size updated to 2MB? This is supposed to be used only for Proxy Handshake. Are the Digest Auth headers large?

Reply:
Digest Auth header is larger than 4096 bytes, in my case, it is about 4150 bytes. The reason to change to 2 * 1024 * 1024 is that I saw the proton impl class TransportImpl has defined BUFFER_RELEASE_THRESHOLD as 2 * 1024 * 1024.

Reply:
Current implementation holds on to the 4k buffers - as long as the Connection is live. With this change, when customers create large no. of Connections - these 2MB buffers are going to show up in the memory profile - which is essentially unused but allocated memory. 

Can you please evaluate these 2 options:
1. Find the maximum size of the buffer needed for Digest Auth. If this is below say 10K then we are good. Just update that constant.
2. If we have to go with 2MB buffers - please explore how you can clear the buffers, once the proxy negotiation is done.

---
In reply to: [261047300](https://github.com/Azure/qpid-proton-j-extensions/pull/10#discussion_r261047300) [](ancestors = 261047300)

Reply:
Thank you for the explanation. I have tried find a way to flush the input socket buffer but seems it is controlled by proton-j lib. So I change to use the 8 * 1024 in our use case now.